engine/apps/schedules/ical_utils.py

```diff
@@ -358,14 +358,14 @@ def list_users_to_notify_from_ical_for_period(
     schedule: "OnCallSchedule",
     start_datetime: datetime.datetime,
     end_datetime: datetime.datetime,
-) -> typing.List["User"]:
+) -> typing.Sequence["User"]:
     users_found_in_ical: typing.Sequence["User"] = []
     events = schedule.final_events(start_datetime, end_datetime)
-    usernames = []
+    usernames: typing.List[str] = []
     for event in events:
         usernames += [u["email"] for u in event.get("users", [])]
 
-    users_found_in_ical = users_in_ical(usernames, schedule.organization)
+    users_found_in_ical = memoized_users_in_ical(tuple(usernames), schedule.organization)
     return users_found_in_ical
```

Comment:
why cast `usernames` to a tuple?
```suggestion
    return memoized_users_in_ical(tuple(usernames), schedule.organization)
```

Reply:
because `memoized_users_in_ical` doesn't work with list, it raises `TypeError: unhashable type: 'list'`